x-pack/libbeat/management/managerV2.go

```diff
@@ -31,15 +38,20 @@ type BeatV2Manager struct {
 
 	logger *logp.Logger
 
-	// Track individual units given to us by the V2 API
-	unitsMut sync.Mutex
-	units    map[string]*client.Unit
-	mainUnit string
+	// track individual units given to us by the V2 API
+	mx      sync.Mutex
```

Comment:
Yeah I can move it. I originally have multiple mutex for different parts but then I would still need to grab the unitsMx so I just changed it to a single mutex because it was safer than causing some deadlock in lock ordering.